yoyodyne/models/base.py

```diff
@@ -11,16 +11,19 @@
 from torch import nn, optim
 
 from .. import batches, defaults, evaluators, schedulers
+from .modules import base, check_encoder_compatibility
```

Comment:
Please don't import symbols, import modules. So you're going to say `import .modules` and then use `modules.base` (or you can separately do `from .modules import base` if you want) and `modules.check_encoder_compatibility`.

Reply:
Oh gotcha. What's the python rationale for that?

Reply:
1. If you import `X` directly it's not easy to figure out where you got it from.
2. Python silently keeps going when you have an import clash so you want to keep things in their original namespace.
3. Each module is its own namespace so if you import `X` directly from `y` in `z.py` then anyone who imports `z` in the same way will have the namespace polluted with `X`. (The exactly same logic is applicable to any C++ style guide.)

Only exception is `__init__.py`, which is supposed to be a header really anyways.